### PR TITLE
Issue/#106/blog-post-description

### PR DESCRIPTION
**Pull request by goostleek**
RSS specification does not restrict _description_ length for RSS items so I assumed maximum length to be 4096 characters in db. Even then we have blog post items that descriptions exceed this limit (the longest description was longer that 10000 chcracters). To avoid `SQLException` in such cases we have to [trim the description](https://github.com/tdziurko/jvm-bloggers/pull/111/files#diff-3debb56cf34039d629e8552c6a684e50R52) to fit `description` column of the `BlogPost` entity.

### CODE REVIEW

File: src/main/java/pl/tomaszdziurko/jvm_bloggers/blog_posts/NewBlogPostStoringActor.java [hunk 1]
```diff
@@ -1,46 +1,57 @@
...
+                .url(postInRss.getLink())
+                .publishedDate(dateToStore.convertDateToLocalDateTime())
+                .approved(rssEntry.getBlog().getDefaultApprovedValue())
+                .blog(rssEntry.getBlog())
+                .build();
+    }
+
+    private void updateDescription(BlogPost blogPost, SyndContent descriptionContent) {
+        if (descriptionContent != null) {
+            final String description = descriptionContent.getValue().abbreviate(BlogPost.MAX_DESCRIPTION_LENGTH);
```
Comment (goostleek) [inline src/main/java/pl/tomaszdziurko/jvm_bloggers/blog_posts/NewBlogPostStoringActor.java]: Trimming description to fit [`BlogPost#description`](https://github.com/tdziurko/jvm-bloggers/pull/111/files#diff-cc7a92da5546ac7e5a7bf75d639eedbfR47) field.

File: src/main/java/pl/tomaszdziurko/jvm_bloggers/blog_posts/NewBlogPostStoringActor.java [hunk 2]
```diff
@@ -1,46 +1,57 @@
...
+
+import org.apache.commons.lang3.StringUtils;
 
 @Slf4j
+@ExtensionMethod({DateTimeUtilities.class, StringUtils.class})
 public class NewBlogPostStoringActor extends AbstractActor {
 
-    private final BlogPostRepository blogPostRepository;
-
     public NewBlogPostStoringActor(BlogPostRepository blogPostRepository) {
```
Comment (cslysy) [inline src/main/java/pl/tomaszdziurko/jvm_bloggers/blog_posts/NewBlogPostStoringActor.java]: For me constructors must be code-free at any cost.

Comment (goostleek) [inline src/main/java/pl/tomaszdziurko/jvm_bloggers/blog_posts/NewBlogPostStoringActor.java]: I do not agree. If constructor were designed to be code-free then why it is even available for programmers? It wouldn't be invented if it were not required. The constructor is for initializing the object and here akka actor initialization takes place. We are defining `PartialFunction` here which is an initialization argument for `Actor`. The code is not executed but passed as an argument. See the [Akka](http://doc.akka.io/japi/akka/2.4.2/akka/japi/pf/ReceiveBuilder.html) you will find more examples how to use Akka framework there.

Comment (cslysy) [inline src/main/java/pl/tomaszdziurko/jvm_bloggers/blog_posts/NewBlogPostStoringActor.java]: By code-free I meant that only object initialziation can take place in constructor whereas businnes logic should be avoided. I didn't noticed that this is object building, not code execution. Of course your code i totaly fine :)

**Comment by goostleek** [comment]
Ok, I removed `@ExtensionMethod`.

To bad it's not supported by _IDEA_ yet :-1: (_Eclipse_ FTW :+1: ). I'm using [`@ExtensionMethod`](https://projectlombok.org/features/experimental/ExtensionMethod.html) and [`@Delegate`](https://projectlombok.org/features/Delegate.html) since I have discovered _Lombok_. It saved me from writing lot of boilerplate code and makes to code easy to read, especially [`@ExtensionMethod`](https://projectlombok.org/features/experimental/ExtensionMethod.html) transforms the code into poetry :)

I'm wondering why Oracle did not adapt this awesome feature into JDK. It's available in [_Groovy_](http://mrhaki.blogspot.com/2013/01/groovy-goodness-adding-extra-methods.html) or [_C#_](https://msdn.microsoft.com/en-us//library/bb383977.aspx) from the very beginning.

**Pull request merged**